core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -191,7 +193,7 @@ private void createZkNamespace() {
     ZkUtils zkUtilsForNamespaceCreation = ZkUtils.apply(
         zkConnForNamespaceCreation,
         zkSessionTimeoutMs, zkSessionTimeoutMs,
-        JaasUtils.isZkSecurityEnabled());
+        JaasUtils.isZkSecurityEnabled() && zKAclsEnabled);
```

Comment:
Is the expected behavior that in the case ZK has SASL enabled, the user might turn ACLs on or off? Do we expect at least some users to enable SASL but disable ACLs?